src/other/index.jl

```diff
@@ -174,9 +174,14 @@ end
     idx
 end
 
-@inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) = getindex(x, collect(Int, idx))
+@inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
+    getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
-@inline Base.getindex(x::AbstractIndex, notidx::Not) = setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, notidx::Not) =
+    setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, idx::Between) = collect(x[idx.first]:x[idx.last])
```

Comment:
I know the existing code does this, but do we really need to call `collect`? Ideally we would just return an `Int` range here above and below.

Reply:
Good point. We do not need this. I have changed it in places where it is safe (there are two cases in `getindex` where still it is better to `collect`).